bootbox.js

```diff
@@ -582,20 +582,28 @@
       let value;
 
       if (options.inputType === 'checkbox') {
-        value = input.find('input:checked').map(function () {
-          return $(this).val();
-        }).get();
+        if(options.required === true) {
+          if(input.find('input:checked').length > 0) {         
+            value = input.find('input:checked').map(function () {
+              return $(this).val();
+            }).get();
+          } 
+          else {
+            // prevents button callback from being called if no checkboxes have been checked
+            return false;
+          }
+        }
+        else {
+          value = input.find('input:checked').map(function () {
+            return $(this).val();
+          }).get();
+        }
```

Comment:
I think it's the same but much more easier to understand:
```suggestion
        value = input.find('input:checked').map(function () {
          return $(this).val();
        }).get();
        if(value.length === 0 && options.required === true) {
            // prevents button callback from being called if no checkboxes have been checked
            return false;
        }
```

Reply:
Agreed. I've merged your suggestion into this PR. Thanks!